app/models/product.rb

```diff
@@ -10,7 +14,12 @@ class Product < ApplicationRecord
   validates :title, :description, :image_url, presence: true
 
   # Adding validation related to price being positive number
-  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  validates :price, 
+    numericality: { 
+      greater_than: :discount_price,
+      message: "must be greater than Discount price"
+    }, 
+    unless: -> (x) { x.price.blank? }
 
   # Validation for title being unique
   validates :title, uniqueness: true
```

Comment:
Can a product be saved with empty title?

Reply:
Handled with
```
  validates :title, :description, :image_url, :price, :discount_price, presence: true
```